modules/statistics/php/stats_behavioural.class.inc

```diff
@@ -63,18 +63,23 @@ class Stats_Behavioural extends \NDB_Form
     {
         parent::setup();
 
-        $DB =& \Database::singleton();
+        $DB   =& \Database::singleton();
+        $user = \User::singleton();
 
-        $centers = $DB->pselect(
-            "SELECT CONCAT('C', CenterID) as ID,
+        $query = "SELECT CONCAT('C', CenterID) as ID,
                     CenterID as NumericID,
                     IFNULL(PSCArea,Name) as LongName,
                     Name as ShortName
-              FROM psc
-              WHERE CenterID <> '1'
-                    AND Study_site = 'Y'",
-            array()
-        );
+                  FROM psc
+                  WHERE CenterID <> '1'
+                        AND Study_site = 'Y'";
+
+        if (!$user->hasPermission('access_all_profiles')) {
+            $site_arr = implode(",", $user->getCenterIDs());
```

Comment:
```suggestion
            $sitesString = implode(",", $user->getCenterIDs());
```

Since this variable isn't an array, the name `site_arr` is misleading. Not a big deal though.